inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -515,6 +541,29 @@ public void writeMessage(InputStream message) {
           statsTraceCtx.outboundMessageSent(outboundSeqNo, -1, -1);
           clientStream.statsTraceCtx.inboundMessage(outboundSeqNo);
           clientStream.statsTraceCtx.inboundMessageRead(outboundSeqNo, -1, -1);
+
+          long messageLength;
```

Comment:
We should do all this messageLength work outside of the `synchronized (this)` block. It's okay to "waste" CPU doing the work and then it turns out `closed == true`.

Reply:
Got it.